validator/src/main/java/com/google/daq/mqtt/util/Common.java

```diff
@@ -30,4 +33,42 @@ static String capitalize(String str) {
     return (str == null || str.isEmpty()) ? str
         : str.substring(0, 1).toUpperCase() + str.substring(1);
   }
+
+  /**
+   * Get the code line where the exception occurred (based on package name).
+   *
+   * @param e         Throwable to trace
+   * @param container containing class to use as package prefix
+   * @return string indicating the offending line
+   */
+  public static String getExceptionLine(Throwable e, Class<?> container) {
```

Comment:
Actually the opposite -- if it's a ValidationException then it's something that much of the code would expect and know how to report intelligently. This is more for the non-ValidationException cases when it would otherwise just be a way-too-detailed expansion of the stack trace. It will work with a ValidationException, just much less needed.

Reply:
Ok great. That is what I was concerned about, i.e. making complicated parsers instead of smart exceptions.